### ISSUE TITLE
selectOneMenu filtering enabled doesn't support NVDA screen reader

### ISSUE THREAD

**Issue by breasteve** [primefaces/primefaces#2177]
## 1) Environment
- PrimeFaces version:
- tested w version 5.3 and 6.0
- tested selectOneMenu with 80 select items and with 7 select items; filter="true" and filter="false"
- Application server + version: WAS 8.5
- Affected browsers: IE 11, Chrome 56

## 2) Expected behavior
selectOneMenu supports NVDA screen reader such that each list item is read as user arrows up/down the list items.

## 3) Actual behavior
list items are never read by NVDA screen reader as user arrows up/down.  The problem seems to be the filter input box control (lstWOBO_focus) has focus and so NVDA screen reader is always reads from it and not from the selected items in list box (where <tr class="... ui-state-highlight...">).  In fact, the entire use of <table> might be problematic with NVDA.

## 4) Steps to reproduce
load test page with selectOneMenu control, start NVDA, tab to selectOneMenu (list box) control, press space bar to open list box, arrow down to various items.  NVDA should read each item label as displayed on the UI.

## 5) Sample XHTML

	<h:panelGrid columns="2" width="580">
		<h:outputLabel id="lblWOBO" for="lstWOBO">Working On Behalf Of:</h:outputLabel>
		<p:selectOneMenu id="lstWOBO" value="#{pc_AjaxListBox.selectedPlayer}" maxlength="30" tabindex="0" 
			style="font-family: arial, Times, serif; font-size: -0.25em; width: 400px;" 
			label="Label here:"
			filter="true" filterMatchMode="contains"
			converter="playerConverter" converterMessage="Failed to convert to Player bean." var="p">	
			<f:selectItem id="selWOBODefault" itemLabel="--- Myself ---" itemValue=""/>		
			<f:selectItems value="#{pc_AjaxListBox.players}" var="player"
				itemLabel="#{player.name}" itemValue="#{player}" />
			<p:column>#{p.name}</p:column>
		</p:selectOneMenu>
	</h:panelGrid>	

## 6) Sample bean

public class AjaxListBox extends PageCodeBase {

       private Player selectedPlayer;
	private List<Player> players;
	
	public AjaxListBox(){
		initialize();
	}
	
	private void initialize(){
		//load players
		if(players == null){
			players = new ArrayList<Player>();			
			for(String s : new AutoComplete().getProviders()){
				players.add(new Player(players.size()+55, s));	//some bogus ID
			}
			Collections.sort(players, new PlayerComparator());			
		}
	}

	public List<Player> getPlayers() {
		return players;
	}

	public void setPlayers(List<Player> players) {
		this.players = players;
	}

	public Player getSelectedPlayer() {
		return selectedPlayer;
	}

	public void setSelectedPlayer(Player selectedPlayer) {
		this.selectedPlayer = selectedPlayer;
	}
}

//---------------------- AutoComplete.java data bean

@ManagedBean(name="autoComplete")
@ApplicationScoped
public class AutoComplete implements Serializable {

	private static final long serialVersionUID = 3115154454834190356L;

public String[] getLocations() {
      String[] l = new String[] {
  			"4569852 - Johnny Marr" 
    		  ,"8953623 - Philip Grubeur" 
    		  ,"7503256 - Keifer Smith"
    		  ,"2401036 - Mario Smithson"
    		  ,"963336 - Kelly Andersonn" 
    		  ,"995021 - Anderson Cooper"
    		  ,"990114 - Derek Rollins"  
    		  ,"330214 - George Armstrong A Long Name (48 char)"
    		  ,"9965220 - Mary Johnson"
    		  ,"1074223 - Albert Collins"
    		  ,"4112583 - Daniel Ewart" 
    		  ,"565201 - Bradley Simon",
  			"772014 - Stewart Glasgow"
    		  ,"663251 - Kimberly Reid"
    		  ,"2004514 - Collin Trefort"
			,"7852014 - Sara Trefort"
			,"6320441 - Sarah Mullins"
			,"9025415 - James Patterson"
			,"305214 - Robert Henry"
			,"3320541 - Henry Dean"
			,"1054855 - Tracy Rollins"
			,"6374410 - Gilbert Morin"
			,"96410522 - Sandy Kofax"
			,"336584 - Walter Johns"
			,"5526499 - Debby Smith"
			,"45101522 - Joshua Rimmer"
			,"96333602 - Peter Glass"
			,"7021455 - Amy Lefebre"
			,"33524811 - Amy Polar"
			,"10548852 - Michelle Morin"
			,"20155521 - Blaine Peters"
			,"8965330 - Gilbert Grape"
			,"1234560 - Tony Rollins"
			,"3456786 - Chad Campbell"
			,"9045991 - Beverly Hills"
			,"444555 - Walter Edney"
			,"890311 - Rupert Dons"
			,"5622204 - Janice Yaurd"
			,"99052144 - Kendry Michaels"
			,"30522014 - Dorn Swallow"
			,"33205214 - Huston Street"
			,"450114522 - Steve Garvey"
			,"8834122 - Albert Ronning"
			,"7845265 - Samuel Smith"
			,"56238559 - Jody Reid"
			,"4401582 - Marg Goldie"
			,"33620544 - Heath Andrews"
			,"3025447 - Tipper Donaldson"
			,"5640122 - Brock Turner"
			,"6302544 - Kelvin Maddock"
			,"205888596 - Mitchell Todd"
			,"234014558 - Xander Bogerts"
			,"662530144 - George Plum"
			,"8934442 - Debby Rey"
			,"CAS3409 - Fanning Mince"
			,"8922103 - Ernie Peppers"
			,"BAK9032 - Gil Martin"
			,"5640122 - Reynolds Anglin"
			,"8845110 - Steve Smithson"
			,"630577411 - Andrew Bogut"
			,"11254855 - Kevin Peters"
			,"6011147 - Jody Vance"
			,"11236695 - Sarah Burning"
			,"1481107 - Tim Meadows"
			,"13669590 - Jim Keith"
			,"7774152 - Keith Richards"
			,"66259444 - Keith Fields"
			,"70842215 - Jordan Henry"
			,"6669990 - Luke Skywalker"
			,"852228 - Dermot Flanders"
			,"33600001 - Joe Smith"
			,"56004152 - Joseph Dean"
			,"2302285 - Daniel Turcott"
			,"56200014 - Wallace Turner"
			,"345902 - Trea Turns"
			,"88834111 - Kyle Kendrick"
			,"90488823 - Donald George"
			,"90322281 - Eddy George"
			,"22228341 - Jordan Falls"
			,"56000148 - Storm Windows"
			,"90338777 - Peter Landings"
			,"11109437 - Broken Knee"  
      };
      Arrays.sort(l);
      return l;
   }

	public List<String> getProviders() {
		List<String> providers = new ArrayList<String>();
		for(String s : getLocations()){
			providers.add(s);
		}
		Collections.sort(providers);
		return providers;
	}
}

//-------------------------  Player.java bean -------------------------------------

public class Player implements Serializable {
	private static final long serialVersionUID = -4027607247939129866L;
	private int id;
	private String name;
	
	public Player(String name) {
		super();
		this.name = name;
	}	

	public Player(int id, String name) {
		super();
		this.id = id;
		this.name = name;
	}

	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
}

//-------------------- converter class -----------------------------------

@FacesConverter("playerConverter")
public class PlayerConverter implements Converter
{

	/**
	 * getAsObject converts String representation to Object instance.
	 */
    @Override
    public Object getAsObject(FacesContext fc, UIComponent uic, String name)
    {    
    	System.out.println("\n\n***************" + this.getClass().getName() + ".getAsObject()...name=" + name);
        return new Player(name);
    }

    /**
     * getAsString converts Object instance to String representation.
     */
    @Override
    public String getAsString(FacesContext fc, UIComponent uic, Object obj)
    {
        if(obj instanceof Player){
            Player p = (Player)obj;
            return p.getName();
        }else if(obj instanceof String){
        	return (String)obj;
        }else{
            StringBuilder sbError = new StringBuilder("The object of class ");
            sbError.append(obj.getClass().getName()).append(" is not of Player");
            throw new ClassCastException(sbError.toString());
        }
    }
}

**Comment by tandraschko** [comment]
It would be great if you could provide a possible solution or even a PR.

**Comment by melloware** [comment]
Duplicate of: https://github.com/primefaces/primefaces/issues/4660  follow that ticket please.
